### PR TITLE
ember sync button

### PR DESCRIPTION
**Pull request by C0rydoras**
- added fetch service
- added sync button to project-detailed

### CODE REVIEW

File: ember/app/components/project-detailed/component.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        this.notification.danger('Project not found');
+      } else if (request.status === 500) {
+        this.notification.danger('An error occurred while syncing the project');
+      }
+    } catch (e) {
+      this.notification.danger(e);
+    }
+    this.loading = false;
+    this.store.findRecord('project', this.args.project.id);
+  }
```
Comment (anehx) [inline ember/app/components/project-detailed/component.js:35]: You can use a task from `ember-concurrency` for this. Among other nice functionality, the task will expose properties like `isRunning` to show a loading state.

Take the time to read about concurrency tasks and then refactor this action to a task.

File: ember/app/components/project-detailed/template.hbs [hunk 2]
```diff
@@ -14,4 +14,10 @@
 
   <hr class='seperator' />
   <DependencyTable @versionedDependencies={{@project.versionedDependencies}} />
+  <UkButton 
+  @label="Sync"
+  @loading={{this.loading}}
+  @disabled={{this.loading}}
+  @onClick={{this.syncProject}}
```
Comment (anehx) [inline ember/app/components/project-detailed/template.hbs:21]: Wrong indentation.

**Review by anehx — CHANGES_REQUESTED** [review]

File: ember/app/components/project-detailed/template.hbs (hunk 2 above)
Comment (C0rydoras) [reply ember/app/components/project-detailed/template.hbs]: fixed: https://github.com/adfinis/Outdated/pull/30/commits/e878e3f0785ab3ab24a4311052fd7e256f48b21c

File: ember/app/components/project-detailed/component.js (hunk 1 above)
Comment (C0rydoras) [reply ember/app/components/project-detailed/component.js]: done here: https://github.com/adfinis/Outdated/pull/30/commits/e878e3f0785ab3ab24a4311052fd7e256f48b21c

**Review by anehx — APPROVED** [review]

**Pull request merged**